PyInstaller/utils/osx.py

```diff
@@ -103,3 +103,59 @@ def fix_exe_for_code_signing(filename):
     ## Write changes back.
     with open(exe_data.filename, 'rb+') as fp:
         exe_data.write(fp)
+
+
+def code_sign(binary, identity, deep=None, verbose=True):
+    """ Code sign binary (.so, .dylib, or .app) using identity.
+
+    Relies on the codesign utility which must be in PATH.
+
+    binary
+        (str) Full path to the binary or .app to be signed.
+    identity
+        (str) The code signing identity to use (as you would give to the
+              ``codesign -s`` option on the command-line).
+    deep
+        (bool) Specify whether to use the codesign --deep option. If not
+               specified, will auto-detect based on whether ``binary`` is a
+               directory (such as a .app or .framework) or a stand-alone file.
+    verbose
+        (bool) Specify whether codesign output should be verbose.
+    """
+    if deep is None and os.path.isdir(binary):
+        deep = True
+    deep = '--deep' if deep else ''
+    verbose = '-v' if verbose else ''
+
+    binary_dir = os.path.dirname(binary)
+    saved_cwd = None
```

Comment:
You can assign to `saved_cwd` directly here. Even if `chdir(binary_dir)` would fail, running `os.chdir(saved_cwd)` in `finally:` does no harm. Or did I miss some case?

(And in the `finally`-claus you can save the `if saved_cwd:` test.)